### ISSUE TITLE
🐛 [2.0.0-beta.6] `biome migrate` does not change `noUnusedVariables` rule options

### ISSUE THREAD

**Issue by wojtekmaj** [biomejs/biome#6239]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.0.2
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         yarn/4.9.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

In Biome v1, `noUnusedVariables` did not cause errors on rest siblings by default. Running `biome migrate` on 2.0.0-beta.6 caused no change to `noUnusedVariables` options and resulted in dozens of warnings across my repository.

Manually changing 

```json
        "noUnusedVariables": "warn"
```

to

```json
        "noUnusedVariables": {
          "level": "warn",
          "options": {
            "ignoreRestSiblings": true
          }
        }
```

restored previous behavior.

### Expected result

`noUnusedVariables` to automatically get `options.ignoreRestSiblings` set to true during migration unless explicitly specified otherwise before migration.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by levrik** [comment]
@ematipico This is still an issue I think. Is the documentation outdated? It says `ignoreRestSibilings` defaults to `true`.

https://biomejs.dev/linter/rules/no-unused-variables/#ignorerestsiblings
